File: front/src/models/Medaille.ts
export type Medaille =
{
    id: number,
    nom: string,
    description: string,
    groupe: number,
    nomFichier: string | null
    nbPoint: number,
    obtentionUnique: boolean
}

export type MedailleRequete =
{
    nom: string,
    description: string,
    groupe: number,
    nbPoint: number
}

export type AttribuerMedailleRequete =
{
    idMedaille: number,
    listeIdPersonnage: number[]
}

File: back/back/Models/Medaille.cs
using LiteDB;

namespace back.Models;

public sealed class Medaille
{
     [BsonId]
     public int Id { get; set; }
     public string Nom { get; set; } = null!;
     public string Description { get; set; } = null!;
     public string? NomFichier { get; set; }
     public int NbPoint { get; set; }

     /// <summary>
     /// 0 => Personnelle, 1 => services, 2 => campagne
     /// </summary>
     public int Groupe { get; set; }
}
